### DISCUSSION TITLE
UX: Command arguments are order sensitive

### DISCUSSION THREAD

**Discussion by jepperson2** [FiloSottile/age#241] [UX reports]
## What were you trying to do
Decrypt using a custom key.

## What happened
I received an error stating `too many arguments`. The problem seemed to be that I had the `-d <filename>` before the `-i <filename>`. Reversing these two arguments caused `age` to behave as I expected. 

The error message as stated made me think the problem was in the `-i <filename>` component of the command as it just says `"input file"`, but I think the problem was that it tried to keep reading more files for the `-d` parameter when I expected it to stop and go on to the `-i` parameter input.

```
$ age -d test.age -i ~/.config/age/keys.txt
Error: too many arguments.
age accepts a single optional argument for the input file.
[ Did age not do what you expected? Could an error be more useful? Tell us: https://filippo.io/age/report ]

$ age -i ~/.config/age/keys.txt -d test.age
hello there
```

**Comment by FiloSottile** [comment]
Thank you for reporting this!

What's going on is that `-d` is not a flag that takes an argument, just a switch, and the input file is an argument so has to go after the flags.

It's a little tricky because the input file is optional, but sounds like we should make this better.

**Comment by sylr** [comment]
Using `github.com/spf13/pflag` instead of the standard `flag` package should help.

**Comment by FiloSottile** [comment]
Better error messages for this is tracked as #160.
